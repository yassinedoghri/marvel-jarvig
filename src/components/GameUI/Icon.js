import { pulse } from "react-animations";
import styled, { keyframes } from "styled-components";

const pulseAnimation = keyframes`${pulse}`;

const Icon = styled.span`
  color: ${props => props.theme.colors.contrastDark.primary};
  padding-left: 0.6em;
  padding-right: 0.3em;
  font-size: 1.8rem;

  & svg {
    animation: ${props =>
      props.animated ? `1s ${pulseAnimation} infinite` : "none"};
  }
`;

export default Icon;
