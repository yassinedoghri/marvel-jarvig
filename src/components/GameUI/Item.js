import styled from "styled-components";

const Item = styled.li`
  display: inline;
  font-size: 1.2rem;

  &:last-child {
    margin-left: 1em;
  }
`;

export default Item;
