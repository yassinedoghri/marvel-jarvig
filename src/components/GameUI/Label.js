import { flash } from "react-animations";
import styled, { keyframes } from "styled-components";

const flashAnimation = keyframes`${flash}`;

const Label = styled.span`
  font-family: Bangers, sans-serif;
  letter-spacing: 0.1em;
  text-shadow: 3px 3px 0 #222, -1px -1px 0 #222, 1px -1px 0 #222,
    -1px 1px 0 #222, 1px 1px 0 #222;
  animation: ${props =>
    props.blink ? `2s ${flashAnimation} infinite` : "none"};
`;

export default Label;
