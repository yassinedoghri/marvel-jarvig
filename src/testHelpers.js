import React from "react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import theme from "theme";
import { ThemeProvider } from "styled-components";

export default function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}
