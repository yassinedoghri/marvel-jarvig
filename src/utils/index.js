export { default as style } from "utils/style";
export { default as baseStyles } from "utils/baseStyles";
export { default as helpers } from "utils/helpers";
export { default as testHelpers } from "utils/testHelpers";
export { default as registerServiceWorker } from "utils/registerServiceWorker";
export { default as media } from "utils/media";
