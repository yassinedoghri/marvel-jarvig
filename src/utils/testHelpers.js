import { createMemoryHistory } from "history";
import React from "react";
import { Provider } from "react-redux";
import {
  ConnectedRouter,
  routerMiddleware,
  routerReducer
} from "react-router-redux";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import rootReducer from "reducers";
import { applyMiddleware, combineReducers, createStore } from "redux";
import { ThemeProvider } from "styled-components";
import theme from "theme";

const history = createMemoryHistory("/");
const routeMiddleware = routerMiddleware(history);

const store = createStore(
  combineReducers({
    ...rootReducer,
    router: routerReducer
  }),
  applyMiddleware(routeMiddleware)
);

export function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export function renderInContext(component) {
  return renderer.create(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}

export function ShallowRenderInContext(component) {
  const renderer = new ShallowRenderer();
  return renderer.render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}
