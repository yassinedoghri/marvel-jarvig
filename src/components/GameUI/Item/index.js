import styled from "styled-components";

import Icon from "components/GameUI/Item/Icon";
import Label from "components/GameUI/Item/Label";

const Item = styled.li`
  display: inline;
  font-size: 1.2rem;

  &:last-child {
    margin-left: 1em;
  }
`;

Item.Icon = Icon;
Item.Label = Label;

export default Item;
