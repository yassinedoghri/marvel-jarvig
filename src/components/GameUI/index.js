import Icon from "components/GameUI/Icon";

import Item from "components/GameUI/Item";
import Label from "components/GameUI/Label";
import styled from "styled-components";

const GameUI = styled.ul`
  padding: 0;
  margin-left: auto;
`;

GameUI.Item = Item;
GameUI.Icon = Icon;
GameUI.Label = Label;

export default GameUI;
