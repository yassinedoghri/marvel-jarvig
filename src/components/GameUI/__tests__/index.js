import GameUI from "components/GameUI";
import React from "react";
import { renderWithTheme } from "utils/testHelpers";

describe("GameUI Block", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(
      <GameUI>
        <GameUI.Item>
          <GameUI.Icon>Icon</GameUI.Icon>
          <GameUI.Label>GameUI Label</GameUI.Label>
        </GameUI.Item>
      </GameUI>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
