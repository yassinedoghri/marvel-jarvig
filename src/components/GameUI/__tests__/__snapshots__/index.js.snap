// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameUI Block renders correctly 1`] = `
<ul
  className="sc-bxivhb dvpYRR"
>
  <li
    className="sc-bwzfXH hAajTB"
  >
    <span
      className="sc-bdVaJa iVzWHn"
    >
      Icon
    </span>
    <span
      className="sc-htpNat kfOksF"
    >
      GameUI Label
    </span>
  </li>
</ul>
`;
